src/app/clusters/scenes-server/SceneTable.h

```diff
@@ -127,148 +127,137 @@ class SceneHandler : public IntrusiveListNodeBase<>
+    }
 
-        SceneTableEntry() = default;
-        SceneTableEntry(SceneStorageId id) : mStorageId(id) {}
-        SceneTableEntry(const SceneStorageId id, const SceneData data) : mStorageId(id), mStorageData(data) {}
-
-        bool operator==(const SceneTableEntry & other) const
-        {
-            return (mStorageId == other.mStorageId && mStorageData == other.mStorageData);
-        }
+    void operator=(const SceneData & other)
+    {
+        SetName(CharSpan(other.mName, other.mNameLength));
+        mExtensionFieldSets    = other.mExtensionFieldSets;
+        mSceneTransitionTimeMs = other.mSceneTransitionTimeMs;
+    }
+};
+} // namespace scene_table_elements
 
-        void operator=(const SceneTableEntry & other)
-        {
-            mStorageId   = other.mStorageId;
-            mStorageData = other.mStorageData;
-        }
-    };
+template <class EFStype>
+class SceneTable
+    : public virtual app::common::FabricTable<scene_table_elements::SceneStorageId, scene_table_elements::SceneData<EFStype>>
+{
+public:
+    using Super = app::common::FabricTable<scene_table_elements::SceneStorageId, scene_table_elements::SceneData<EFStype>>;
+    using SceneTableEntry = typename Super::TableEntry;
+    using SceneStorageId  = scene_table_elements::SceneStorageId;
+    using SceneData       = scene_table_elements::SceneData<EFStype>;
 
     SceneTable(){};
 
     virtual ~SceneTable(){};
 
-    // Not copyable
-    SceneTable(const SceneTable &) = delete;
-
-    SceneTable & operator=(const SceneTable &) = delete;
-
-    virtual CHIP_ERROR Init(PersistentStorageDelegate * storage) = 0;
-    virtual void Finish()                                        = 0;
-
     // Global scene count
-    virtual CHIP_ERROR GetEndpointSceneCount(uint8_t & scene_count)                         = 0;
-    virtual CHIP_ERROR GetFabricSceneCount(FabricIndex fabric_index, uint8_t & scene_count) = 0;
+    inline CHIP_ERROR GetEndpointSceneCount(uint8_t & scene_count) { return this->GetEndpointEntryCount(scene_count); }
+    inline CHIP_ERROR GetFabricSceneCount(FabricIndex fabric_index, uint8_t & scene_count)
+    {
+        return this->GetFabricEntryCount(fabric_index, scene_count);
+    }
 
     // Data
-    virtual CHIP_ERROR GetRemainingCapacity(FabricIndex fabric_index, uint8_t & capacity)                                   = 0;
-    virtual CHIP_ERROR SetSceneTableEntry(FabricIndex fabric_index, const SceneTableEntry & entry)                          = 0;
-    virtual CHIP_ERROR GetSceneTableEntry(FabricIndex fabric_index, SceneStorageId scene_id, SceneTableEntry & entry)       = 0;
-    virtual CHIP_ERROR RemoveSceneTableEntry(FabricIndex fabric_index, SceneStorageId scene_id)                             = 0;
-    virtual CHIP_ERROR RemoveSceneTableEntryAtPosition(EndpointId endpoint, FabricIndex fabric_index, SceneIndex scene_idx) = 0;
+    inline CHIP_ERROR SetSceneTableEntry(FabricIndex fabric_index, const SceneTableEntry & entry)
```

Comment:
So this is changing from an abstract interface (with SceneTableImpl as a possible, but not only, implementation) to a concrete implementation, right?  Was this discussed with @lpbeliveau-silabs ?

Reply:
No longer an issue with removal of `FabricTable`, this interface stays ~the same